plugins/inputs/huebridge/huebridge_test.go

```diff
@@ -0,0 +1,95 @@
+package huebridge
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/tdrn-org/go-hue/mock"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestConfig(t *testing.T) {
+	conf := config.NewConfig()
+	err := conf.LoadConfig("testdata/conf/huebridge.conf")
+	require.NoError(t, err)
+	require.Len(t, conf.Inputs, 1)
+	plugin, ok := conf.Inputs[0].Input.(*HueBridge)
+	err = plugin.Init()
+	require.NoError(t, err)
+	require.True(t, ok)
+	require.NotNil(t, plugin.Bridges)
+	require.Len(t, plugin.Bridges, 4)
+	require.NotEmpty(t, plugin.RemoteClientId)
+	require.NotEmpty(t, plugin.RemoteClientSecret)
+	require.NotEmpty(t, plugin.RemoteCallbackUrl)
+	require.NotEmpty(t, plugin.RemoteTokenDir)
+	require.NotNil(t, plugin.RoomAssignments)
+	require.Len(t, plugin.RoomAssignments, 2)
+	require.Equal(t, config.Duration(60*time.Second), plugin.Timeout)
+	require.Equal(t, "secret", plugin.TLSKeyPwd)
+	require.True(t, plugin.InsecureSkipVerify)
+}
+
+func TestInitBridges(t *testing.T) {
+	// Start mock server
+	bridgeMock := mock.Start()
+	require.NotNil(t, bridgeMock)
+	defer bridgeMock.Shutdown()
+	// Prepare token dir
+	tokenDir, err := os.MkdirTemp("", "TestInitBridges")
+	require.NoError(t, err)
+	tokenFile := filepath.Join(tokenDir, mock.MockClientId, strings.ToUpper(mock.MockBridgeId)+".json")
+	bridgeMock.WriteTokenFile(tokenFile)
```

Comment:
Please use [`t.TempDir()`](https://pkg.go.dev/testing#T.TempDir) here which ensures that the directory gets removed after the test (no matter if it failed or succeeded).

Reply:
Adapted with latest commit 3249fb1a6ff458f46d44dc7da5210b73675e1a44